### DISCUSSION TITLE
Still getting error "The following signatures couldn't be verified because the public key is not available" even after renewing keys

### DISCUSSION THREAD

**Discussion by duffysk** [oerdnj/deb.sury.org#1800] [General]
Hey there,

I'm running Debian buster and hitting following error
`Err:6 https://packages.sury.org/php buster InRelease
  The following signatures couldn't be verified because the public key is not available: NO_PUBKEY B188E2B695BD4743`

I removed key, added again, but hitting the same issue over and over again. Any advice please?

Thanks!

Rado

**Comment by oerdnj** [comment]
You have to show us what exactly you did to "renew" the keys if you want our help.

**Reply by duffysk** [reply]
This is what I did:
`rm /etc/apt/trusted.gpg.d/php.gpg`
`apt-key del B188E2B695BD4743`
`wget -O /etc/apt/trusted.gpg.d/php.gpg https://packages.sury.org/php/apt.gpg`

**Comment by oerdnj** [comment]
Does the sha256sum matches?
```
$ sha256sum /etc/apt/trusted.gpg.d/php.gpg
b3ea944563435e54bb64f181ee8bc26200985d09164cdc4c1702fc3ef051f19d  /etc/apt/trusted.gpg.d/php.gpg
```

And how did you add the repository and what is the contents of `cat /etc/apt/sources.list.d/php.list` (or how it is named on your system)?

**Reply by duffysk** [reply]
It does. Here is it:
```
root@apache:~# sha256sum /etc/apt/trusted.gpg.d/php.gpg
b3ea944563435e54bb64f181ee8bc26200985d09164cdc4c1702fc3ef051f19d  /etc/apt/trusted.gpg.d/php.gpg
```
php.list contains:
`deb [signed-by=/etc/apt/trusted.gpg.d/php.gpg] https://packages.sury.org/php/ buster main`

**Reply by oerdnj** [reply]
Well, I don't think that `[signed-by=/etc/apt/trusted.gpg.d/php.gpg]` comes from my instructions, the current instructions look like this: `deb [signed-by=/usr/share/keyrings/deb.sury.org-php.gpg] https://packages.sury.org/php/ bullseye main`

See https://packages.sury.org/php/README.txt

Before it just accepted any key.

So, what instructions did you follow installing the repository?

But most probably, you have the same key with the expired signature stored elsewhere on the system and you are replacing just a single copy, but the `apt-key` is finding the expired one first.

Try `apt-key list` and also check the `/usr/share/keyrings/` for any cruft.

**Reply by duffysk** [reply]
I did follow your instructions, essentially https://packages.sury.org/php/README.txt but as it did not work, I changed the path (value) of signed-by parameter, but the behavior is exactly the same.

When trying `apt-key list` the only key having sury.org in it is
```
/etc/apt/trusted.gpg.d/php.gpg
------------------------------
pub   rsa3072 2019-03-18 [SC] [expires: 2024-02-16]
      1505 8500 A023 5D97 F5D1  0063 B188 E2B6 95BD 4743
uid           [ unknown] DEB.SURY.ORG Automatic Signing Key <deb@sury.org>
sub   rsa3072 2019-03-18 [E] [expires: 2024-02-16]
```
I did not find anything suspicious in `/usr/share/keyrings/`

```
root@apache:/usr/share/keyrings# ls
debian-archive-bullseye-automatic.gpg           debian-archive-buster-automatic.gpg           debian-archive-keyring.gpg            debian-archive-stretch-security-automatic.gpg  debian-nonupload.gpg  salt-archive-keyring.gpg
debian-archive-bullseye-security-automatic.gpg  debian-archive-buster-security-automatic.gpg  debian-archive-removed-keys.gpg       debian-archive-stretch-stable.gpg              debian-role-keys.gpg
debian-archive-bullseye-stable.gpg              debian-archive-buster-stable.gpg              debian-archive-stretch-automatic.gpg  debian-maintainers.gpg                         deb.sury.org-php.gpg
```

**Reply by oerdnj** [reply]
> I did follow your instructions ... I changed the path 

So..., you didn't. And you failed to mention those changes that you did to your system. Has it occurred to you are just wasting people's time by withholding and hiding the information?

Since you did random changes to your system that you only know about you and my patience asking you questions about the changes just ran out you are on your own.

I will follow this up only if you are able to reproduce this on a clean fully-updated system, following the instruction to the letter.

**Reply by duffysk** [reply]
This "So..., you didn't. And you failed to mention those changes that you did to your system. Has it occurred to you are just wasting people's time by withholding and hiding the information?" is not accurate as either way it does not work for me. When I rolled the change back, the result is the same so I don't think I'm wasting anyone's time or hiding the information.
